test/jdk/jdk/classfile/BoundAttributeTest.java

```diff
@@ -67,4 +72,33 @@ void testReadMethodParametersAttributeWithoutParameterName() {
         List<MethodParameterInfo> parameters = assertDoesNotThrow(methodParametersAttribute::parameters);
         assertTrue(parameters.get(0).name().isEmpty());
     }
+
+    @Test
+    void testBadEntryTypeInList() {
+        var cf = ClassFile.of();
+
+        // Craft an attribute list with index to badly typed attributes
+        var bytes = cf.build(ClassDesc.of("Test"), clb -> {
+            var cp = clb.constantPool();
+            var oneClassString = cp.utf8Entry("Test$Ape");
+            var oneClass = cp.classEntry(oneClassString);
+            ((DirectClassBuilder) clb).writeAttribute(new UnboundAttribute.AdHocAttribute<>(Attributes.nestMembers()) {
+                @Override
+                public void writeBody(BufWriter b) {
```

Comment:
```suggestion
                public void writeBody(BufWriterImpl b) {
```
This is a conflict with #20205, whichever is integrated later should update this part.